src/components/Timeline/Event/Events.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types'
+
+import ErrorMessage from 'components/elements/ErrorMessage'
+import LoadingMessage from 'components/elements/LoadingMessage'
+import Highlight from 'components/elements/Highlight'
+import EventCard from 'components/Timeline/Event/EventCard'
+import BootstrapPlaybook from 'components/Timeline/Playbook/BootstrapPlaybook'
+import * as eventTypeActions from 'actions/eventTypeActions'
+
+export const Events = ({events, ticketId, incidentId}) => <div>
+  {Array.from(events)
+    .map(event =>
+      <ConnectedEvent
+        key={event.id}
+        event={event}
+        ticketId={ticketId}
+      />)
+  }
+</div>
+
+Events.propTypes = {
+  events: PropTypes.array.isRequired,
+  ticketId: PropTypes.string.isRequired,
+  incidentId: PropTypes.number
+}
+
+const eventHasValidDisplayText = (event) =>
+  event && event.data && event.data.DisplayText
+
+export const Event = ({
+  eventTypeIsFetching,
+  eventTypeIsError,
+  event,
+  ticketId
+}) => eventTypeIsFetching && !eventHasValidDisplayText(event)
+? <LoadingMessage
+  message={'Fetching Event Type Information'}
+  actionForRetry={eventTypeActions.fetchEventType(event.eventTypeId)}
+/>
+: eventTypeIsError && !eventHasValidDisplayText(event)
+  ? <ErrorMessage
+    message={`Error fetching eventType: ${event.eventTypeId}`}
+    actionForRetry={eventTypeActions.fetchEventType(event.eventTypeId)}
+  />
+  : <DisplayEvent
+    event={event}
+    ticketId={ticketId}
+  />
+
+export const mapStateToEventProps = (state, ownProps) => {
+  const { event, ticketId } = ownProps
+
+  return {
+    eventTypeIsFetching: state.eventTypes.fetching.includes(event.eventTypeId),
+    eventTypeIsError: state.eventTypes.error.includes(event.eventTypeId),
+    event,
+    ticketId
+  }
+}
+
+export const ConnectedEvent = connect(mapStateToEventProps)(Event)
```

Comment:
[⎂](tktktk) This could be inlined to the scope which it is used in. About 50 lines above it.

Reply:
Rearranged this file so that Events comes immediately after ConnectedEvent